src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java

```diff
@@ -33,135 +33,236 @@
  * Checks for restricted tokens beneath other tokens.
  * </p>
  * <p>
- * Examples of how to configure the check:
+ * WARNING: This is a very powerful and flexible check, but, at the same time,
+ * it is low-level and very implementation-dependent because its results depend
+ * on the grammar we use to build abstract syntax trees. Thus we recommend using
+ * other checks when they provide the desired functionality. Essentially, this
+ * check just works on the level of an abstract syntax tree and knows nothing
+ * about language structures.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code limitedTokens} - Specify set of tokens with limited occurrences as descendants.
+ * Default value is {@code {}}.
+ * </li>
+ * <li>
+ * Property {@code minimumDepth} - Specify the minimum depth for descendant counts.
+ * Default value is {@code 0}.
+ * </li>
+ * <li>
+ * Property {@code maximumDepth} - Specify the maximum depth for descendant counts.
+ * Default value is {@code java.lang.Integer.MAX_VALUE}.
+ * </li>
+ * <li>
+ * Property {@code minimumNumber} - Specify a minimum count for descendants.
+ * Default value is {@code 0}.
+ * </li>
+ * <li>
+ * Property {@code maximumNumber} - Specify a maximum count for descendants.
+ * Default value is {@code java.lang.Integer.MAX_VALUE}.
+ * </li>
+ * <li>
+ * Property {@code sumTokenCounts} - Control whether the number of tokens found
+ * should be calculated from the sum of the individual token counts.
+ * Default value is {@code false}.
+ * </li>
+ * <li>
+ * Property {@code minimumMessage} - Define the error message when the minimum count is not reached.
+ * Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code maximumMessage} - Define the error message when the maximum count is exceeded.
+ * Default value is {@code null}.
+ * </li>
+ * </ul>
+ * <p>
+ * Switch with no default:
```

Comment:
It seems weird seeing this text above the example configuration because it is not fully clear "this is how you configure the check to produce a violation for this specific thing". 
Can we add wording we use in other checks to the start of this configuration description and the others.
Example: `To configure the check to produce a violation on a switch statement with no default case:`

Reply:
done.